packages/psss-server/src/routes/alerts/FetchAlertsRoute.ts

```diff
@@ -5,38 +5,68 @@
 
 import { Route } from '../Route';
 
-const REPORT_BY_STATUS = {
-  archived: 'PSSS_Archived_Alerts',
+const CATEGORY_TO_REPORT = {
+  archive: 'PSSS_Archived_Alerts',
   active: 'PSSS_Active_Alerts',
 };
 
-type Status = keyof typeof REPORT_BY_STATUS;
+type AlertCategory = keyof typeof CATEGORY_TO_REPORT;
 
-function validateStatus(status: string): asserts status is Status {
-  const statuses = Object.keys(REPORT_BY_STATUS);
-  if (!statuses.includes(status)) {
-    throw new Error(`Invalid alert status ${status}, must be one of ${statuses}`);
+type AlertReportRow = {
+  organisationUnit: string;
+  period: string;
+  syndrome: string;
+};
+
+function validateStatus(category: string): asserts category is AlertCategory {
+  const categories = Object.keys(CATEGORY_TO_REPORT);
+  if (!categories.includes(category)) {
+    throw new Error(`Invalid alert category '${category}', must be one of ${categories}`);
   }
 }
 
 export class FetchAlertsRoute extends Route {
   async buildResponse() {
-    const { startWeek, endWeek, orgUnitCodes } = this.req.query;
-    const { status } = this.req.params;
-
-    validateStatus(status);
-    const reportCode = REPORT_BY_STATUS[status];
+    const { startWeek, endWeek } = this.req.query;
 
-    const reportData = await this.reportConnection?.fetchReport(
-      reportCode,
-      orgUnitCodes?.split(','),
-      [startWeek, endWeek],
+    const alertData: AlertReportRow[] = await this.fetchAlertData();
+    const data = await Promise.all(
+      alertData.map(async alert => {
+        const confirmedData = await this.fetchConfirmedDataForAlert(alert);
+        return { ...alert, ...confirmedData };
+      }),
     );
 
     return {
       startWeek,
       endWeek,
-      data: reportData,
+      data,
     };
   }
+
+  private fetchAlertData = async () => {
+    const { startWeek, endWeek, orgUnitCodes } = this.req.query;
+    const { category } = this.req.params;
+
+    validateStatus(category);
+    const alertReport = CATEGORY_TO_REPORT[category];
+    const { results } = await this.reportConnection.fetchReport(
+      alertReport,
+      orgUnitCodes?.split(','),
```

Comment:
Wonder if `fetchReport` should be responsible for doing the split... seems like I see this code a lot... but no biggie 👍

Reply:
It is true that at the moment, we are splitting the string input in the routes, and then joining it again in the connection, which would seem "redundant" at a first glance.

On the other hand, I like the fact that communication between `psss-server` classes is done using TS objects, even if it means that we eventually go from `A => B => A`. It probably makes for cleaner interfaces. Then, the only places were objects are (de)stringified are in the package's "seams" (input and output of the package)